flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/CatalogManager.java

```diff
@@ -1775,8 +1783,39 @@ public ResolvedCatalogView resolveCatalogView(CatalogView view) {
         if (view instanceof ResolvedCatalogView) {
             return (ResolvedCatalogView) view;
         }
+
+        if (view instanceof QueryOperationCatalogView) {
+            return new ResolvedCatalogView(
+                    view, view.getUnresolvedSchema().resolve(schemaResolver));
+        }
+
+        final List<Operation> parse = parser.parse(view.getExpandedQuery());
```

Comment:
What is the error behavior when the parse fails. Should we wrap a exception around it. E.g. if a UDF is not available anymore or the referenced table changed.